packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -619,6 +757,86 @@ enum SelectionStatus {
   none,
 }
 
+/// The details of a selection.
+///
+/// This includes information such as the status of the selection indicating
+/// if it is collapsed or uncollapsed, the start and end offsets of the selection
+/// local to the [SelectionListener] that reports this object, and whether
+/// the selection is ongoing.
+///
+/// This object is created by callers of the [SelectionListenerSelectionChangedCallback] callback.
+///
+/// See also:
+///
+///   * [SelectionListener], which provides a [SelectionDetails] object
+///     for the selection under its subtree in its [SelectionListener.onSelectionChanged]
+///     callback.
+@immutable
+final class SelectionDetails with Diagnosticable {
+  /// Creates a selection details object.
+  const SelectionDetails({
+    required this.status,
+    required this.selectionFinalized,
+    required this.localStartOffset,
+    required this.localEndOffset,
+  });
+
+  /// The status of the selection under the [Selectable] or [SelectionHandler]
+  /// that created this object.
+  final SelectionStatus status;
+
+  /// Whether the selection is ongoing.
+  ///
+  /// This is false if the selection is ongoing and true if the selection
+  /// is finalized.
+  ///
+  /// A selection is ongoing in scenarios like an ongoing mouse drag,
+  /// long press drag, or in between states where both selection edges
+  /// have not reached their final position.
+  final bool selectionFinalized;
+
+  /// The offset where the selection starts.
+  ///
+  /// This is relative to the [Selectable] subtree of the object creator.
```

Comment:
Nit: "object creator" => "creator of this [SelectionDetails] instance"

Also below for localStartOffset.